src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemory.java

```diff
@@ -406,6 +408,24 @@ public void setEntryFailureMessage(GerritTriggeredEvent event, Run r, String fai
     }
 
     /**
+     * Creates a snapshot clone of the current coordination memory status.
+     *
+     * @return the report
+     */
+    @Nonnull
+    public synchronized BuildMemoryReport report() {
+        BuildMemoryReport report = new BuildMemoryReport();
+        for (Map.Entry<GerritTriggeredEvent, MemoryImprint> entry : memory.entrySet()) {
+            List<Entry> triggered = new LinkedList<Entry>();
```

Comment:
@rsandell But then they show straightforward benchmarks of common use cases. 

The only cases where you see a major performance benefit for LinkedLists vs ArrayLists are where you repeatedly trigger large ArrayCopy operations (for example, insertion at the middle or beginning, and inserts at beginning may be reversed).

There's a reason the ArrayDeque (circular array impl) is recommended over Stack/Queue/LinkedList versions and some sources describe it as a bug to use the older classes in that way.  But then again I'm an optimization nerd...